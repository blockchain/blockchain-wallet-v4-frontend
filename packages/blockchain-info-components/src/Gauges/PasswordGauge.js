import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 3px;
  bottom: 0;
  left: 0;
`
const Bar = styled.div`
  display: inline-flex;
  width: 20%;
  height: 100%;
  box-sizing: border-box;
  background-color: ${props => props.theme[props.color]};
`

const selectColor = score => {
  switch (score) {
    case 1:
      return 'error'
    case 2:
      return 'red'
    case 3:
      return 'brand-yellow-lighter'
    case 4:
      return 'brand-yellow'
    case 5:
      return 'success'
    default:
      return 'transparent'
  }
}

const PasswordGauge = props => {
  const { score } = props
  const color = selectColor(score)

  return (
    <Container>
      {score > 0 ? <Bar color={color} /> : <Bar color='transparent' />}
      {score > 1 ? <Bar color={color} /> : <Bar color='transparent' />}
      {score > 2 ? <Bar color={color} /> : <Bar color='transparent' />}
      {score > 3 ? <Bar color={color} /> : <Bar color='transparent' />}
      {score > 4 ? <Bar color={color} /> : <Bar color='transparent' />}
    </Container>
  )
}

PasswordGauge.propTypes = {
  score: PropTypes.oneOf([1, 2, 3, 4, 5]).isRequired
}

PasswordGauge.defaultProps = {
  score: 0
}

export default PasswordGauge
