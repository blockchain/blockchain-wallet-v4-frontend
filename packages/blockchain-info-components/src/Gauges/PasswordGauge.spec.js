import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import PasswordGauge from './PasswordGauge'

describe('PasswordGauge component', () => {
  it('default renders correctly', () => {
    const component = shallow(<PasswordGauge score={1} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
