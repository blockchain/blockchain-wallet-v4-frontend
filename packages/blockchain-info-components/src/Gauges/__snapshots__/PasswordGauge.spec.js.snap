// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordGauge component default renders correctly 1`] = `
<PasswordGauge__Container>
  <PasswordGauge__Bar
    color="error"
  />
  <PasswordGauge__Bar
    color="transparent"
  />
  <PasswordGauge__Bar
    color="transparent"
  />
  <PasswordGauge__Bar
    color="transparent"
  />
  <PasswordGauge__Bar
    color="transparent"
  />
</PasswordGauge__Container>
`;
